test/integration/snapshots-in-forks.test.js

```diff
@@ -0,0 +1,44 @@
+import Microcosm from '../../src/microcosm'
+
+describe('Snapshots', function() {
+  it('forks accommodate actions that were pushed before the parent was forked', function() {
```

Comment:
This test specifically covers an edge case I found where:

1. You create a presenter
2. That presenter pushes an action in the setup method
3. A react router route embeds another presenter as a child

This put Microcosm into a state where the fork form the child presenter did not have a snapshot for the parent. The prototypal inheritance update addresses this problem.